x-pack/plugins/security_solution/public/cases/components/all_cases/index.tsx

```diff
@@ -341,9 +341,9 @@ export const AllCases = React.memo<AllCasesProps>(
 
     const onTableRowClick = useMemo(
       () =>
-        memoize<(id: string) => () => void>((id) => () => {
+        memoize<(theCase: Case) => () => void>((theCase) => () => {
```

Comment:
this is really confusing to have a useMemo with memoize, I think we can simplify everything by doing that 
```
const tableRowProps = useCallback(
      (item) => {
        const rowProps = {
          'data-test-subj': `cases-table-row-${item.id}`,
        };

        if (isModal && onRowClick) {
          return {
            ...rowProps,
            onClick: onRowClick(item),
          };
        }

        return rowProps;
      },
      [isModal, onRowClick]
    );
```

Reply:
This is a trick to have a memorized version of the callback along with the item. `onClick: onRowClick(item)` will not work. You need to do `onClick: () => onRowClick(item)` which will create an anonymous new function on every render and will create unnecessary rerenders.

Reply:
Yes I did not read that correctly but we can still simplify it like that if you look at the type in eui for EuiBasicTable
```
const onTableRowClick = useCallback(
      (item: Case) => {
        if (onRowClick) {
          onRowClick(item.id, item.title);
        }
      },
      [onRowClick]
    );

    const tableRowProps = useCallback(
      ({ id, title }) => ({
        'data-test-subj': `cases-table-row-${id}`,
        ...(isModal ? { onClick: onTableRowClick } : {}),
      }),
      [isModal, onTableRowClick]
    );
```